src/github/pullRequestManager.ts

```diff
@@ -785,24 +785,30 @@ export class PullRequestManager implements IPullRequestManager {
 			});
 	}
 
-	async getPullRequestChangedFiles(pullRequest: IPullRequestModel): Promise<Github.PullRequestsGetFilesResponseItem[]> {
-		Logger.debug(`Fetch changed files of PR #${pullRequest.prNumber} - enter`, PullRequestManager.ID);
+	async getPullRequestFileChangesInfo(pullRequest: IPullRequestModel): Promise<IRawFileChange[]> {
+		Logger.debug(`Fetch file changes, base, head and merge base of PR #${pullRequest.prNumber} - enter`, PullRequestManager.ID);
 		const { octokit, remote } = await (pullRequest as PullRequestModel).githubRepository.ensure();
 
-		let response = await octokit.pullRequests.getFiles({
-			owner: remote.owner,
+		const { data } = await octokit.repos.compareCommits({
 			repo: remote.repositoryName,
-			number: pullRequest.prNumber,
-			per_page: 100
+			owner: remote.owner,
+			base: `${pullRequest.base.repositoryCloneUrl.owner}:${pullRequest.base.ref}`,
+			head: `${pullRequest.head.repositoryCloneUrl.owner}:${pullRequest.head.ref}`
 		});
-		let { data } = response;
 
-		while (response.headers.link && octokit.hasNextPage(response.headers)) {
-			response = await octokit.getNextPage(response.headers);
-			data = data.concat(response.data);
+		pullRequest.mergeBase = data.merge_base_commit.sha;
+
+		if (!pullRequest.base) {

```

Comment:
when is `pullRequest.base`  undefined? should this be before the `compareCommits` call?

Reply:
thanks, I already pushed a fix for that. `base` and `head` are missing when we use the search api.